php/libraries/Database.class.inc

```diff
@@ -165,11 +165,20 @@ class Database
 
         }
 
-        $this->_PDO = new PDO(
-            "mysql:host=$host;dbname=$database;charset=UTF8",
-            $username,
-            $password
-        );
+        try {
+            $this->_PDO = new PDO(
+                "mysql:host=$host;dbname=$database;charset=UTF8",
+                $username,
+                $password
+            );
+        } catch (PDOException $e) {
+            error_log(
+                'Could not establish PDO connection using supplied '
+                .' credentials. Please correct the configuration settings in '
+                .' project/config.xml to valid MySQL credentials.'
+            );
+            exit(1);
```

Comment:
The database class (or any class) should never call `exit`. This should throw a new PDOException with a different message rather than using error_log and calling exit..

Reply:
tested this, it works, will retest after this change^